Modules/main.c

```diff
@@ -288,6 +288,9 @@ pymain_run_module(const wchar_t *modname, int set_argv0)
         return pymain_exit_err_print();
     }
     result = PyObject_Call(runmodule, runargs, NULL);
+    if (!result && PyErr_Occurred() == PyExc_KeyboardInterrupt) {
```

Comment:
Maybe accepting subclasses would be better:

```suggestion
    if (!result && PyErr_ExceptionMatches(PyExc_KeyboardInterrupt)) {
```

But I see that run_eval_code_obj() also tests using ``==``. Either leave it as it is, or change both functions.

I would be interesting to also modify pymain_err_print(), to also catch KeyboardInterrupt there.

Maybe a new _Py_HandleKeyboardInterrupt() function would be useful to always handle KeyboardInterrupt the same way.

Reply:
I'm not following why run_eval_code_obj isn't run from PyObject_Call?

Reply:
Let's not worry about subclasses.

I think Victor is saying that there are a lot of exit paths from pymain_run_module() and we only special case KeyboardInterrupt in one of them (the main one, though). But we can't move the check for KeyboardInterrupt into pymain_err_print() because that function doesn't receive the result.

The only thing I realize we should probably add is setting `UnhandledKeyboardInterrupt = 0;` before the PyObject_Call() call, to match what's done in run_eval_code_obj() (see the big comment at the start there).